packages/adp-tooling/src/preview/routes-handler.ts

```diff
@@ -170,6 +170,75 @@ export default class RoutesHandler {
         }
     };
 
+    /**
+     * Handler for retrieving existing controller extension data from the workspace.
+     *
+     * @param req Request
+     * @param res Response
+     * @param next Next Function
+     */
+    public handleGetControllerExtensionData = async (req: Request, res: Response, next: NextFunction) => {
+        try {
+            const params = req.params as { controllerName: string };
+            const controllerName = sanitize(params.controllerName);
+            const codeExtFiles = await this.readAllFilesByGlob('/**/changes/*_codeExt.change');
+
+            let controllerExists = false;
+            let controllerPath = '';
+            let controllerPathFromRoot = '';
+
+            const project = this.util.getProject();
+            const sourcePath = project.getSourcePath();
+            const rootPath = project.getRootPath();
+            const projectName = rootPath.split(/[\\/]/).slice(-1)[0];
+
+            const getPath = (projectPath: string, fileName: string) =>
+                path.join(projectPath, FolderNames.Changes, FolderNames.Coding, fileName).replace(/[\\/]/g, path.sep);
```

Comment:
Why is the `replace` needed?

Reply:
Because the path that is returned by path.join will look like this on windows: `C:\\Users\\projects\\adp.project`. And I am replacing those slashes with platform-specific path separator character for the current operating system, like:

Windows: `\` (backslash)
Unix-like systems (Linux, macOS): `/` (forward slash)

The path will be visible in the dialog, and this `C:\\Users\\projects\\adp.project` does not look good. :)

Reply:
Changed in: [1a92ecb](https://github.com/SAP/open-ux-tools/pull/1408/commits/1a92ecb1bdc70de9df0f9f07579f230c5f72f366)

Reply:
Great, thanks!